test/es-module/test-esm-tla-unfinished.mjs

```diff
@@ -72,24 +72,31 @@ describe('ESM: unsettled and rejected promises', { concurrency: !process.env.TES
   });
 
   it('should exit for an unsettled TLA promise with warning', async () => {
-    const { code, stderr, stdout } = await spawnPromisified(execPath, [
+    const { code, stderr } = await spawnPromisified(execPath, [
       fixtures.path('es-modules/tla/unresolved.mjs'),
     ]);
 
-    assert.match(stderr, /Warning: Detected unsettled top-level await at.+unresolved\.mjs:1/);
+    assert.match(stderr, /Warning: Detected unsettled top-level await at.+unresolved\.mjs:5/);
     assert.match(stderr, /await new Promise/);
-    assert.strictEqual(stdout, '');
+    assert.strictEqual(code, 13);
+  });
+
+  it('the process exit event should provide the correct code for an unsettled TLA promise', async () => {
+    const { code, stdout } = await spawnPromisified(execPath, [
+      fixtures.path('es-modules/tla/unresolved.mjs'),
+    ]);
+
+    assert.strictEqual(stdout, 'the exit listener received code: 13\n');
     assert.strictEqual(code, 13);
```

Comment:
```suggestion
    const { code, stdout, stderr } = await spawnPromisified(execPath, [
      fixtures.path('es-modules/tla/unresolved.mjs'),
    ]);

    assert.deepStrictEqual({ code, stdout, stderr }, {
      code: 13,
      stdout: 'the exit listener received code: 13\n',
      stderr: '',
    });
```

Reply:
this is related to [my comment above](https://github.com/nodejs/node/pull/56872/files#r1976447793)  the `'es-modules/tla/unresolved.mjs'` fixture is being used in two tests, one (the previous test) in which we test its standard error and one (this one) where we test its standard output so I wouldn't really go with this suggestion, which would also fail since `stderr` is not empty

However if you think that this testing structure is confusing the two tests can be combined into one, I'm happy to do that if you think the current structure is problematic

Reply:
I don't think it's a problem to test something twice: better twice than none, and it can give very precious info when debugging a failing test.